packages/wonder-blocks-button/src/components/button-core.tsx

```diff
@@ -331,170 +329,148 @@ export const _generateStyles = (
     }
 
     const padding =
-        size === "large" ? theme.padding.xLarge : theme.padding.large;
+        size === "large" ? theme.root.padding.large : theme.root.padding.medium;
 
-    const disabledState = theme.color[kind].disabled;
+    const borderWidthKind = theme.root.border.width[kind];
+    const outlineOffsetKind = theme.root.border.offset[kind];
+    const themeVariant = theme.root.color[kind][actionType];
+    const disabledState = theme.root.color[kind].disabled;
 
-    const disabledStateStyles = {
+    const disabledStatesStyles = {
         borderColor: disabledState.border,
+        borderWidth: borderWidthKind.default,
         background: disabledState.background,
         color: disabledState.foreground,
     };
 
-    let newStyles: Record<string, CSSProperties> = {};
-    if (kind === "primary") {
-        const themeColorAction = theme.color.primary[actionType];
-
-        const sharedFocusHoverStyling = {
-            outlineOffset: theme.border.offset.primary,
-            outlineStyle: "solid",
-            outlineWidth: theme.border.width.focused,
-        };
-
-        const focusStyling = {
-            ...sharedFocusHoverStyling,
-            outlineColor: themeColorAction.focus.border,
-        };
-
-        const hoverStyling = {
-            ...sharedFocusHoverStyling,
-            outlineColor: themeColorAction.hover.border,
-        };
-
-        const activePressedStyling = {
-            background: themeColorAction.press.background,
-            outlineColor: themeColorAction.press.border,
-            outlineOffset: theme.border.offset.primary,
-            outlineStyle: "solid",
-            outlineWidth: theme.border.width.focused,
-        };
-
-        newStyles = {
-            default: {
-                background: themeColorAction.default.background,
-                color: themeColorAction.default.foreground,
-                paddingInline: padding,
-                ":hover": hoverStyling,
-                ":focus-visible": focusStyling,
-                ":active": activePressedStyling,
+    const newStyles = {
+        default: {
+            borderRadius: theme.root.border.radius[size],
+            paddingInline: kind === "tertiary" ? 0 : padding,
+            // theming
+            borderStyle: "solid",
+            borderWidth: borderWidthKind.default,
+            borderColor: themeVariant.default.border,
+            background: themeVariant.default.background,
+            color: themeVariant.default.foreground,
+
+            /**
+             * States
+             *
+             * Defined in the following order: hover, active, focus.
```

Comment:
Should the comment say: `focus, hover, active`? I see the focusStyles are applied first!

Reply:
I've reverted back the order to have focus defined last.